src/parser.ts

```diff
@@ -7876,10 +7876,7 @@ export function parseClassDeclaration(
 
   const { tokenValue } = parser;
 
-  if (
-    ((parser.token & 0b0000000000000000001_0000_11111111) ^ 0b0000000000000000000_0000_01010100) >
-    0b0000000000000000001_0000_00000000
-  ) {
+  if (parser.token & Token.Keyword && parser.token !== Token.ExtendsKeyword) {
```

Comment:
Please view these two lines.

Reply:
Sure this will work?  Why not  

`if (parser.token & (Token.IsIdentifier | Token.FutureReserved) && parser.token !== Token.ExtendsKeyword ) {`

Keyword will make this case valid `class switch() {}`.

The two token can be merged into signed integer (32-bit), but maybe better to keep it as it is for readability?

Reply:
I didn't quite understand the existing line, it is confusing for me. The existing unit tests didn't complain about my code. I will add your test case and try again.

Reply:
The throw is inside the if branch, my code still invalids `class switch() {}`, `class eval {}` `class arguments {}`.
```js

  if (parser.token & Token.Keyword && parser.token !== Token.ExtendsKeyword) {
    if (isStrictReservedWord(parser, context, parser.token)) {
      report(parser, Errors.UnexpectedStrictReserved);
    }

    if ((parser.token & Token.IsEvalOrArguments) === Token.IsEvalOrArguments) {
      report(parser, Errors.StrictEvalArguments);
    }
```

```
huocp@chunpengs-mbp ~/g/meriyah (logic-assign)> npx ts-node
> import {parseModule} from './src/meriyah'
{}
> var r = parseModule('class switch() {}')
[/Users/huocp/github/meriyah/src/errors.ts:187
    throw new ParseError(parser.index, parser.line, parser.column, type, ...params);
    ^

Uncaught SyntaxError: [1:12]: Unexpected strict mode reserved word
] {
  index: 12,
  line: 1,
  column: 12,
  description: '[1:12]: Unexpected strict mode reserved word',
  loc: { line: 1, column: 12 }
}
> 
```

Added test.

Reply:
This confused me: identifier is also considered a keyword?

```ts
  Keyword                 = 1 << 12,
  Contextual              = 1 << 13 | Keyword,

  IsIdentifier             = 1 << 17 | Contextual,
```

Reply:
@KFlash can you review the 2 posts above?

Reply:
That it throws on invalid case is correct.  The bitwise masks seems weird.  Contextual should be future reserved I guess or plain ident.

Reply:
merge this code?

Reply:
you can